### PR TITLE
Fix: activity regularizer not normalized by batch size

### PR DESCRIPTION
**Pull request by Abhinavexists**
fixes: #22018 

The documentation in `keras/src/regularizers/regularizers.py` states that activity regularizer losses are divided by batch size to ensure consistent regularization strength across different batch sizes. 

However, the implementation was not performing this normalization, causing activity regularization strength to scale with batch size.

### CODE REVIEW

File: keras/src/layers/layer.py [hunk 1]
```diff
@@ -974,7 +975,17 @@ def maybe_convert(x):
...
+                            loss = self.activity_regularizer(output)
+                            if output.ndim > 0:
+                                # Normalize by batch size to ensure consistent
+                                # regularization strength across batch sizes
+                                batch_size = ops.cast(
+                                    ops.shape(output)[0], dtype=loss.dtype
+                                )
+                                loss = ops.where(
+                                    batch_size > 0, loss / batch_size, 0.0
+                                )
```
Comment (hertschuh) [inline keras/src/layers/layer.py]: Just do `ops.divide_no_nan(loss, batch_size)`, it should be faster.

Comment (Abhinavexists) [reply keras/src/layers/layer.py]: updated with divide_no_nan
